bin/expo-setup.js

```diff
@@ -19,28 +19,24 @@ if (!isPlainObject(appJson) || !isPlainObject(appJson.expo)) {
   );
 }
 
-appJson.expo.experiments ||= {};
-// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- Set to empty object in previous line, if falsy
-appJson.expo.experiments.typedRoutes = true;
-
-appJson.expo.plugins = [
-  [
-    'expo-build-properties',
-    {
-      ios: {
-        newArchEnabled: true,
-      },
-      android: {
-        newArchEnabled: true,
-      },
-    },
-  ],
-];
-
-await writeFile(appFilePath, JSON.stringify(appJson, null, 2), 'utf8');
-console.log(
-  '✅ Enabled New Architecture and typedRoutes experiment in app.json',
-);
+const expoConfig =
+  `import { ExpoConfig } from "expo/config";
+
+const config: ExpoConfig = ${JSON.stringify(appJson.expo, null, 2)
+    .replace(/"([^"]+)":/g, '$1:')
+    .replace(/"(.*?)"/g, `'$1'`)
+    .replace(/([}\]])(\s*[}\]])/g, '$1,$2')
+    .replace(/}(\s+\])/g, '},$1')
+    .replace(/(?<!,)(\n\s*[}\]])/g, ',$1')};
+
+export default config;
+`.trim() + '\n';
+
+await writeFile('app.config.ts', expoConfig, 'utf8');
+console.log('✅ Converted app.json to app.config.ts');
```

Comment:
since we're taking over converting to `app.config.ts`, we need to have a code comment in this code to point to an Expo issue or PR that `create-expo-app` / `create-expo` (mention both names in the issue / PR if you create it) should be the config format that is generated - active maintainers of `create-expo-app` can be tagged on this issue / PR (maybe Simek is active, not sure)